dvc/repo/experiments/utils.py

```diff
@@ -273,9 +274,23 @@ def resolve_name(
     return result
 
 
+def is_valid_name_format(name: str) -> bool:
+    # https://github.com/git/git/blob/master/Documentation/git-check-ref-format.txt
+    invalid_chars = set("/\\ @^~:*?[]\177")  # \177 is DEL
+    return not any(
+        [
+            bool(invalid_chars & set(name)),
+            ".." in name,
+            name.startswith("."),
+            name.endswith((".lock", ".")),
+            re.search(r"[\x00-\x1F]", name) is not None,  # ascii control characters
+        ]
+    )
```

Comment:
```suggestion
return scm.check_ref_format(str(ref)) or "/" in ref.name
```

Reply:
Is there a specific reason for us to redefine the git regex check here as opposed to just using the git check as-is?

Reply:
The pygit2 backend function exec's the actual git plumbing shell command, which isn't ideal in the context of studio. https://github.com/iterative/studio/pull/7825#issuecomment-1746063875

Reply:
I'm assuming you mean gitpython? pygit2 wraps libgit2 which doesn't use CLI git at all. 

DVC defaults to using pygit2 and dulwich, `scm.check_ref_format` should not be hitting the gitpython backend at all unless studio is also using some other scmrepo calls that force the use of gitpython (scmrepo will prefer the most recently used backend)

Reply:
You are correct then. I'll have to check and see if something else strange is going on then; as I see in debug logs it executes the plumbing command.

Reply:
off the top of your head do you know which calls force the use of gitpython?

Reply:
@dacbd, is that happening during clone? Studio uses `gitpython` for cloning.

Reply:
None of the native `scm.func()` calls should require the use of gitpython. 

The issue is probably that studio still uses gitpython directly with both `scm.gitpython.func()` calls and with `import git; git.func()` calls (which will all execute CLI git from the shell).

If enforcing the use of some preferred scmrepo backend is required in studio, it would b e better for studio to just explicitly do this check itself with
```
if scm.pygit2.check_ref_format(name) or "/" in name:
```
(or `scm.dulwich.check_ref_format()...`)